VideoClassification/test.py

```diff
@@ -0,0 +1,99 @@
+"""Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Defines the testing procedure.
+"""
+import os
+import time
+import tensorflow as tf
+import tensorflow_datasets as tfds
+import tensorflow.keras.metrics as metrics
+import numpy as np
+
+import readers
+import reader_utils
+import model as model_lib
+import loss
+
+def test_model(model, data_reader, test_dir, batch_size):
+	"""Test the model on test dataset attained from test_dir.
+	
+	Args:
+		model: the trained keras model
+		data_reader: the TfRecords datareader used to load training and validation datasets
+		test_dir: string representing the directory containing the test TfRecords files
+	Returns:
+		eval_dict: dictionary containing important evaulation metrics
+	"""
+	#Create evaluation metrics
+	auc_calculator = metrics.AUC(multi_label=True)#, curve='PR')
+	pr_calculator = metrics.PrecisionAtRecall(0.7)
+	rp_calculator = metrics.RecallAtPrecision(0.7)
+
+	#Prepare data
+	batch_num = 0
+	test_dataset = data_reader.get_dataset(test_dir, batch_size=batch_size, type="validate")
+	test_dataset = tfds.as_numpy(test_dataset)
+
+	for batch in test_dataset:
+		test_input = tf.convert_to_tensor(batch[0])
+		test_labels = tf.convert_to_tensor(batch[1])
+
+		predictions = model.predict(test_input)
+
+		loss_val = loss.custom_crossentropy(test_labels, predictions)
+
+		#Update Metrics
+		auc_calculator.update_state(test_labels, predictions)
+		#pr_calculator.update_state(test_labels, predictions)
```

Comment:
We may uncomment them in the submitted code

Reply:
Yes, I have rectified this issue